waggle-dance-extensions/src/main/java/com/hotels/bdp/waggledance/extensions/client/ratelimit/BucketKeyGenerator.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (C) 2016-2024 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.waggledance.extensions.client.ratelimit;
+
+public class BucketKeyGenerator {
+
+  private final String prefix;
+
+  public BucketKeyGenerator(String prefix) {
+    this.prefix = prefix;
+  }
+
+  public String generateKey(String key) {
+    if (prefix != null && !prefix.isEmpty()) {
+      return prefix + "_" + key;
+    }
+    return "" + key;
```

Comment:
Why not just return key? 🤔

Reply:
See the test, it prevents from us returning null as a key instead it will return "null" which is more in line with the prefix + null key => "prefix_null". It's super minor probably won't matter.